python/pyarrow/_dataset_parquet.pyx

```diff
@@ -631,7 +752,13 @@ cdef class ParquetFileWriteOptions(FileWriteOptions):
             coerce_timestamps=None,
             allow_truncated_timestamps=False,
             use_compliant_nested_type=True,
+            encryption_config=None,
         )
+        IF PARQUET_ENCRYPTION_ENABLED:
+            if self._properties["encryption_config"] is not None:
+                print("Encryption is not enabled in this build of pyarrow. "
+                      "Please reinstall pyarrow with encryption enabled.")
+
```

Comment:
Two questions here: (1) Am I reading this wrong or is this logic not quite right? It doesn't look like this will print if someone tries to pass `encryption_config` w/ a build of PyArrow not built with encryption enabled and (2) should this be a print or a throw? Considering a user might miss this message and the potential consequences of thinking you've encrypted your data when you haven't, maybe this should throw instead.

Reply:
Also, let's make sure this is tested to avoid any regressions.

Reply:
Looks good, thank you.